flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -37,9 +39,24 @@
 @PublicEvolving
 public class ExecutionConfigOptions {
 
+	// ------------------------------------------------------------------------
+	//  State Options
+	// ------------------------------------------------------------------------
+
+	@Documentation.TableOption(execMode = Documentation.ExecMode.STREAMING)
+	public static final ConfigOption<Duration> IDLE_STATE_RETENTION =
+		key("table.exec.state.ttl")
+			.durationType()
+			.defaultValue(Duration.ofMillis(0))
+			.withDescription("A time-to-live (TTL) can be assigned to the keyed state of any type. " +
+				"If a TTL is configured and a state value has expired, " +
+				"the stored value will be cleaned up on a best effort basis. " +
+				"Default value is 0, which means that it will never clean up state.");
```

Comment:
```suggestion
			.withDescription("Specifies a minimum time interval for how long idle state " +
					"(i.e. state which was not updated), will be retained. State will never be " +
					"cleared until it was idle for less than the minimum time, and will be cleared " +
					"at some time after it was idle. Default is never clean-up the state.\n" +
					"NOTE: Cleaning up state requires additional overhead for bookkeeping.\n" +
					"Default value is 0, which means that it will never clean up state.");
```